pkg/server/apiHandler.go

```diff
@@ -104,62 +91,13 @@ func (h *apiHandler) register(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
 		return
 	}
-	msg := &register.ZRegisterMsg{}
-	if err := proto.Unmarshal(b, msg); err != nil {
-		log.Printf("Failed to parse register message: %v", err)
-		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
-		return
-	}
-	serial := msg.Serial
-	err = h.manager.OnboardCheck(onboardCert, serial)
-	if err != nil {
-		_, invalidCert := err.(*common.InvalidCertError)
-		_, invalidSerial := err.(*common.InvalidSerialError)
-		_, usedSerial := err.(*common.UsedSerialError)
-		switch {
-		case invalidCert, invalidSerial:
-			log.Printf("failed authentication %v", err)
-			http.Error(w, err.Error(), http.StatusUnauthorized)
-		case usedSerial:
-			log.Printf("used serial %v", err)
-			http.Error(w, err.Error(), http.StatusConflict)
-		default:
-			log.Printf("Error checking onboard cert and serial: %v", err)
-			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
-		}
-		return
-	}
-	// the passed cert is base64 encoded PEM. So we need to base64 decode it, and then extract the DER bytes
-	// register the new device cert
-	certPemBytes, err := base64.StdEncoding.DecodeString(string(msg.PemCert))
+	status, err := registerProcess(h.manager, b, onboardCert)
```

Comment:
I see no issue with it working, but what do we gain from moving `register()`'s logic out to `registerProcess()`? is it just to make the files smaller, but no other material change?

Reply:
If so, it would have been better to do it as a separate PR. These are unrelated changes - simplifying the codebase and adding v2 - and should be in separate PRs. We can leave it for this one, but we really should keep these separate if we can. If there are other reasons that it had to be in here, so be it.

Reply:
The main reason here and there is to avoid code duplicates between apiV1 and apiV2 handlers. It seems reasonable for me to do it inside the PR.

Reply:
As you explained below, that makes a lot of sense. I didn't clearly understand that on first reading. Looks good.